src/mixins/amalgamation-mixin.ts

```diff
@@ -315,6 +326,79 @@ export default class AmalgamationMixin extends Vue {
     const promises = this.getAmalgamatingBusinesses.map(fetchTingInfo)
     const amalgamatingBusinesses = await Promise.all(promises)
     this.setAmalgamatingBusinesses(amalgamatingBusinesses)
+
+    // return the holding/primary business (or undefined)
+    return this.getAmalgamatingBusinesses.find(business =>
+      (business.role === AmlRoles.HOLDING || business.role === AmlRoles.PRIMARY)
+    )
```

Comment:
This is kind of a side-effect (which is normally undesirable) and yet, useful every time this method is called. It saves having to look for holding/primary business in other code.